docs/design/collaboration/Building the collaboration paradigm between App Developers and Platform Developers with Kusion.md

```diff
@@ -0,0 +1,235 @@
+
+- Application engineers mainly focus on business logic and choose infrastructures to support them in achieving their business goals.
+- Platform engineers are responsible for infrastructure standardization which helps to reduce the cognitive load of application engineers and meet their requirements. This, in turn, boosts the productivity of the organization.
+
+We aim to minimize application developers' cognitive load and enable self-service so we do not let application developers care about low-level infrastructure details. They only need to declare what feature they want in a developer-friendly way, such as declaring one application needs a MySQL database and public Internet access with Kusion models provided by platform developers. Then Kusion will provision a MySQL database and a LoadBalancer that complies with platform standardization.
+
+We introduce a set of Kusion models to standardize deployment and the root model is AppConfiguration. AppConfiguration is a declarative application configuration model maintained by application developers to configure all operation intentions during SLCD and more details can be found on our [website](https://kusionstack.io/docs/user_docs/concepts/appconfiguration). Each Kusion model in the AppConfiguration represents a building block built by platform developers to standardize the application's behavior and hide infrastructure complexity, such as database, networking, and monitoring.
+
+## Terms
+
+Before we dive into details, it is necessary to clarify certain terms.
+- Environment: An environment is a logical concept representing a space where applications can be deployed. It could be a stage during your deployment such as `dev`,`staging`, and `prod` or cloud vendors such as `AWS`,`Azure`, and `Aliyun` or a combination of the two.
+- [Project](https://kusionstack.io/docs/user_docs/concepts/glossary#project): A project is a logical concept to bundle application configurations.
+- [Stack](https://kusionstack.io/docs/user_docs/concepts/glossary#stack): A stack is a logical concept represents where you execute a Kusion command.
+
+## Platform developer's workflow
+### Design Kusion models
+A Kusion model is a building block designed by platform developers to standardize application deployments and it consists of three parts: 
+
+- App developer-oriented schema: Fields in this schema are recommended to be understandable to application developers and environment-agnostic. For example, a database Kusion model schema only contains fields like database engine type and database version.
+- Kusion model inputs: Model inputs contain fields used to standardize applications but are varied from different environments. The subnets in the `prod` and `dev` of one application are usually different and Kusion model inputs can help platform developers change these platform standardizations easily.
```

Comment:
I wonder if conceptually we should include module inputs as part of the module.
You could have different inputs for different env/cloud/etc, but they should be fed into the same module.